jupyterlab_gcedetails/src/components/hardware_scaling_status.tsx

```diff
@@ -105,46 +112,101 @@ export class HardwareScalingStatus extends React.Component<Props, State> {
     window.addEventListener('beforeunload', this.preventPageClose);
   }
 
-  async componentDidUpdate() {
-    const { hardwareConfiguration, notebookService, onCompletion } = this.props;
-    const { machineType, gpuType, gpuCount, attachGpu } = hardwareConfiguration;
+  async componentDidUpdate(prevProps, prevState) {
```

Comment:
This method has gotten pretty large and it's starting to feel like maybe we need to refactor to avoid conflating the two concerns of managing the state transitions from one operation to the next with the UX presentation of the current state. We can discuss in our standup.

Reply:
I've resolved this for the time being by handling displaying the error page in the hardware scaling status component rather than the dialog. I've also cleaned up the code a bit by using methods to make it clear what is going on in each case of this method. In a future PR I think we should consider moving towards using a state management system like redux as we discussed in our meeting to separate state management and UX concerns.